src/spyglass/utils/mixins/export.py

```diff
@@ -0,0 +1,315 @@
+            restr_str = "True"  # otherwise stored in table as '1'
+
+        if isinstance(restr_str, str) and "SELECT" in restr_str:
+            raise RuntimeError(
+                "Export cannot handle subquery restrictions. Please submit a "
+                + "bug report on GitHub with the code you ran and this"
+                + f"restriction:\n\t{restr_str}"
+            )
+
+        if isinstance(restr_str, str) and len(restr_str) > 2048:
+            raise RuntimeError(
+                "Export cannot handle restrictions > 2048.\n\t"
+                + "If required, please open an issue on GitHub.\n\t"
+                + f"Restriction: {restr_str}"
+            )
+
+        if isinstance(restr_str, str):
+            restr_str = bash_escape_sql(restr_str, add_newline=False)
+
+        if restr_str in self._export_cache[self.full_table_name]:
+            return
+        self._export_cache[self.full_table_name].add(restr_str)
+
+        self._export_table.Table.insert1(
+            dict(
+                export_id=self.export_id,
+                table_name=self.full_table_name,
+                restriction=restr_str,
+            )
+        )
+        restr_logline = restr_str.replace("AND", "\n\tAND").replace(
+            "OR", "\n\tOR"
+        )
+        logger.debug(f"\nTable: {self.full_table_name}\nRestr: {restr_logline}")
+
+    def _log_fetch_nwb(self, table, table_attr):
+        """Log fetch_nwb for export table."""
+        tbl_pk = "analysis_file_name"
+        fnames = self.fetch(tbl_pk, log_export=True)
+        logger.debug(
+            f"Export: fetch_nwb\nTable:{self.full_table_name},\nFiles: {fnames}"
+        )
+        self._export_table.File.insert(
+            [{"export_id": self.export_id, tbl_pk: fname} for fname in fnames],
+            skip_duplicates=True,
+        )
+        fnames_str = "('" + "', ".join(fnames) + "')"  # log AnalysisFile table
+        table()._log_fetch(restriction=f"{tbl_pk} in {fnames_str}")
+
+    def _run_with_log(self, method, *args, log_export=True, **kwargs):
+        log_this_call = FETCH_LOG_FLAG.get()  # One log per fetch call
+        if log_this_call and not self.database == "common_usage":
+            FETCH_LOG_FLAG.set(False)
+        try:
+            ret = method(*args, **kwargs)
+        finally:
+            if log_this_call:
+                FETCH_LOG_FLAG.set(True)
+        if log_export and self.export_id and log_this_call:
+            if getattr(method, "__name__", None) == "join":
+                other = kwargs.get("other")
+                if not hasattr(other, "_log_fetch"):
+                    logger.warning(f"Cannot export log join for\n{other}")
+                    __import__("pdb").set_trace()
+                restriction = (  # Fetch self as restricted by other
+                    self.proj()
+                    .join(other.proj(), log_export=False)
+                    .fetch(*self.primary_key, as_dict=True)
+                )
```

Comment:
Do we need to run `log_fetch` on `other` in order to log the entries from that table as well?

Reply:
This might make the assumption that `other` is restricted in some way. If so, the parent restriction process should be caught, or a parent `fetch` process would be caught, or it would be caught on up cascade. So, the edge case is maybe

- Tables: A -> B -> C
- `(B & restr) * C`, without fetch, ignores C in the export

Reply:
Agreed.  Since there's already a case for join here, we could expand it to include log the matched elements from `other`

eg.
```python
for table in [self, other]:
    restriction = (  # Fetch restriction of joined table
        self.proj()
        .join(other.proj(), log_export=False)
        .fetch(*table.primary_key, as_dict=True)
    )
    for r in restriction:
        table._log_fetch(restriction=r)
```